controllers/servicediscovery_controller_unit_test.go

```diff
@@ -49,10 +49,6 @@ func serviceDiscoveryUnitTestsReconcileNormal() {
 		}
 		_, err := reconciler.ReconcileNormal(ctx, controllerCtx.GuestClusterContext)
 		Expect(err).NotTo(HaveOccurred())
-
-		// Update the VSphereCluster and its status in the fake client.
-		Expect(controllerCtx.ControllerContext.Client.Update(ctx, controllerCtx.VSphereCluster)).To(Succeed())
-		Expect(controllerCtx.ControllerContext.Client.Status().Update(ctx, controllerCtx.VSphereCluster)).To(Succeed())
```

Comment:
Similar to above - the object is being checked in place in this test so no need to update it in the fake client.